### PR TITLE
Better logging for "Produced attestation..."

### PR DESCRIPTION
**Pull request by terenc3t**
Add validator public key for `Produced attestation..` log, makes it easier to tell who produced attestation

### CODE REVIEW

File: validator/client/validator_attest.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
 		"slot":           slot - params.BeaconConfig().GenesisSlot,
 	}).Infof("%v Attesting to beacon chain head...", truncatedPk)
 
-	log.Infof("Produced attestation with block root: %#x", attestation.Data.BeaconBlockRootHash32)
+	log.Infof("%v Produced attestation with head root: %#x", truncatedPk, attestation.Data.BeaconBlockRootHash32)
```
Comment (prestonvanloon) [inline validator/client/validator_attest.go]: Please use withField

File: validator/client/validator_propose.go [hunk 2]
```diff
@@ -140,8 +140,8 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, idx string) {
 	)
 	log.WithFields(logrus.Fields{
 		"blockRoot":       fmt.Sprintf("%#x", blkResp.BlockRootHash32),
-		"validator":       truncatedPk,
 		"numAttestations": len(block.Body.Attestations),
 		"numDeposits":     len(block.Body.Deposits),
+		"validator":       truncatedPk,
```
Comment (terenc3t) [inline validator/client/validator_propose.go]: to be consistent with `attester` where `pub key` is always the last field

File: validator/client/validator_attest.go [hunk 3]
```diff
@@ -126,26 +126,25 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
 	// TODO(#1366): Use BLS to generate an aggregate signature.
 	attestation.AggregateSignature = []byte("signed")
 
-	log.WithField(
```
Comment (terenc3t) [inline validator/client/validator_attest.go]: dont need this, already logging `attData.BeaconBlockRootHash32` below

File: validator/client/validator_attest.go [hunk 4]
```diff
@@ -126,26 +126,25 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
...
-	).Info("Current beacon chain head block")
 	log.WithFields(logrus.Fields{
-		"justifiedEpoch": attData.JustifiedEpoch - params.BeaconConfig().GenesisEpoch,
-		"shard":          attData.Shard,
-		"slot":           slot - params.BeaconConfig().GenesisSlot,
+		"shard": attData.Shard,
+		"slot":  slot - params.BeaconConfig().GenesisSlot,
 	}).Infof("%v Attesting to beacon chain head...", truncatedPk)
 
-	log.Infof("Produced attestation with block root: %#x", attestation.Data.BeaconBlockRootHash32)
```
Comment (terenc3t) [inline validator/client/validator_attest.go]: don't need this, already logging `attestation.Data.BeaconBlockRootHash32` below

File: validator/client/validator_attest.go [hunk 5]
```diff
@@ -126,26 +126,25 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
...
 	attResp, err := v.attesterClient.AttestHead(ctx, attestation)
 	if err != nil {
 		log.Errorf("Could not submit attestation to beacon node: %v", err)
 		return
 	}
 	log.WithFields(logrus.Fields{
-		"attestationHash": fmt.Sprintf("%#x", attResp.AttestationHash),
+		"headHash":        fmt.Sprintf("%#x", attData.BeaconBlockRootHash32),
```
Comment (terenc3t) [inline validator/client/validator_attest.go]: here we use the proper names for attestations, `head`, `source` and `target`

File: validator/client/validator_attest.go [hunk 6]
```diff
@@ -126,26 +126,25 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
...
-	log.WithField(
-		"blockRoot", fmt.Sprintf("%#x", attData.BeaconBlockRootHash32),
-	).Info("Current beacon chain head block")
 	log.WithFields(logrus.Fields{
-		"justifiedEpoch": attData.JustifiedEpoch - params.BeaconConfig().GenesisEpoch,
-		"shard":          attData.Shard,
-		"slot":           slot - params.BeaconConfig().GenesisSlot,
+		"shard": attData.Shard,
+		"slot":  slot - params.BeaconConfig().GenesisSlot,
 	}).Infof("%v Attesting to beacon chain head...", truncatedPk)
```
Comment (prestonvanloon) [inline validator/client/validator_attest.go]: Please use `Info` without formatting, Put the pk as a field.

File: validator/client/validator_attest.go [hunk 7]
```diff
@@ -126,26 +126,25 @@ func (v *validator) AttestToBlockHead(ctx context.Context, slot uint64, idx stri
...
+		"headHash":        fmt.Sprintf("%#x", attData.BeaconBlockRootHash32),
+		"sourceHash":      fmt.Sprintf("%#x", attData.JustifiedBlockRootHash32),
+		"targetHash":      fmt.Sprintf("%#x", attData.EpochBoundaryRootHash32),
+		"slot":            attData.Slot - params.BeaconConfig().GenesisSlot,
 		"shard":           attData.Shard,
-		"slot":            slot - params.BeaconConfig().GenesisSlot,
-	}).Infof("%v successfully attested", truncatedPk)
+		"attestationHash": fmt.Sprintf("%#x", attResp.AttestationHash),
+		"validator":       truncatedPk,
+	}).Info("%v Attested latest head")
```
Comment (prestonvanloon) [inline validator/client/validator_attest.go]:
```suggestion
	}).Info("Attested latest head")
```

**Pull request merged**